src/storage/mvcc/txn.rs

```diff
@@ -191,29 +191,29 @@ impl<S: Snapshot> MvccTxn<S> {
 
         if value.is_some() && is_short_value(value.as_ref().unwrap()) {
             self.lock_key(key, lock_type, primary.to_vec(), options.lock_ttl, value);
-        } else {
+        } else if value.is_some() {
```

Comment:
The readability of this snippet is worse, i prefer keep the original code, since in most cases the value is some.